dj/typing.py

```diff
@@ -46,27 +50,133 @@ class SQLADialect(Protocol):  # pylint: disable=too-few-public-methods
 Stream = Iterator[Row]
 
 
-class ColumnType(str, Enum):
+PRIMITIVE_TYPES = {
+    "BYTES",
+    "STR",
+    "FLOAT",
+    "INT",
+    "DECIMAL",
+    "BOOL",
+    "DATETIME",
+    "DATE",
+    "TIME",
+    "TIMEDELTA",
+    "NULL",
+    "WILDCARD",
+}
+
+COMPLEX_TYPES = {"LIST": 1, "DICT": 2}
```

Comment:
I think `ARRAY` instead of `LIST` is the more common term with databases. I checked postgres, spark, and trino and they all use that.

Reply:
Agree. Since we need to work with Spark and Trino for starters I would recommend `ARRAY` and `MAP` for your first two complex types. ROW (or named tuple) should be next in line.

Reply:
added row and changed the other names.